ui/pages/notifications/notification-components/types/notifications/notifications.ts

```diff
@@ -0,0 +1,56 @@
+import type { FC } from 'react';
+import type { Notification } from '../../../../../../app/scripts/controllers/metamask-notifications/types/notification/notification';
+
+/**
+ * NotificationFC is the shared component interface for all notification components
+ */
+type NotificationFC<N = Notification> = FC<{
+  notification: N;
+  onClick?: () => void;
+}>;
+
+type BodyOnChainNotification<N = Notification> = {
+  type: 'body_onchain_notification';
```

Comment:
Why does body and footer need this type?

We can just add this to the details. and discriminate from there?

Reply:
nit-picky, this is working fine.